### ISSUE TITLE
[ERROR] appsignal: Error in minutely probe 'sidekiq': ERR unknown command `connection`, with args beginning with:  

### ISSUE THREAD

**Issue by Aeramor** [appsignal/appsignal-ruby#891]
If your bug is about how AppSignal works in your app specifically we recommend you contact us at support@appsignal.com with your bug report instead.

---

## Describe the bug
Just getting spam that says "appsignal: Error in minutely probe 'sidekiq': ERR unknown command `connection`, with args beginning with:  " over and over again

## To Reproduce

- Using AppSignal for Ruby gem version 3.1.5
- sidekiq 7.0, rails 7.0.4

No code to run, fails non-stop

**Comment by unflxw** [comment]
Thanks for the bug report @Aeramor! We'll be looking into this shortly.

**Comment by tombruijn** [comment]
@Aeramor For me it fails even sooner on minutely probe start up, with this error:

```
[2022-11-07T13:49:00 (process) #230][ERROR] Error while initializing minutely probe 'sidekiq': uninitialized constant Redis

        Gem::Version.new(::Redis::VERSION) >= Gem::Version.new("3.3.5")
                         ^^^^^^^
[2022-11-07T13:49:00 (process) #230][DEBUG] /app/vendor/bundle/ruby/3.1.0/gems/appsignal-3.1.5/lib/appsignal/probes/sidekiq.rb:11:in `dependencies_present?'
```

Looks like sidekiq changed their redis dependency to redis-client, named `RedisClient`, hence the name error.

I'm not sure I'm reproducing this properly yet. Or do you have the `redis` gem in your app's Gemfile as well?

**Comment by tombruijn** [comment]
If I remove the version check the failed in my last comment I get a similar error as originally reported:

```
[2022-11-07T14:26:22 (process) #435][ERROR] Error in minutely probe 'sidekiq': ERR unknown command `connection`, with args beginning with:
[2022-11-07T14:26:22 (process) #435][DEBUG] /app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.11.1/lib/redis_client/connection_mixin.rb:22:in `call'
/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:200:in `block (2 levels) in call'
/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.11.1/lib/redis_client/middlewares.rb:16:in `call'
/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:199:in `block in call'
/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:616:in `ensure_connected'
/app/vendor/bundle/ruby/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:198:in `call'
/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.1/lib/sidekiq/redis_client_adapter.rb:25:in `method_missing'
/integration/lib/appsignal/probes/sidekiq.rb:92:in `block in hostname'
/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.1/lib/sidekiq/config.rb:156:in `block in redis'
/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:65:in `block (2 levels) in with'
/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:64:in `handle_interrupt'
/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:64:in `block in with'
/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:61:in `handle_interrupt'
/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:61:in `with'
/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.1/lib/sidekiq/config.rb:153:in `redis'
/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.1/lib/sidekiq.rb:73:in `redis'
/integration/lib/appsignal/probes/sidekiq.rb:92:in `hostname'
/integration/lib/appsignal/probes/sidekiq.rb:78:in `gauge'
/integration/lib/appsignal/probes/sidekiq.rb:52:in `track_stats'
/integration/lib/appsignal/probes/sidekiq.rb:31:in `call'
/integration/lib/appsignal/minutely.rb:140:in `block (3 levels) in start'
/integration/lib/appsignal/minutely.rb:137:in `each'
/integration/lib/appsignal/minutely.rb:137:in `block (2 levels) in start'
/integration/lib/appsignal/minutely.rb:134:in `loop'
/integration/lib/appsignal/minutely.rb:134:in `block in start'
```

Looks like we'll have to update the probe to work with the redis and redis-client gems.

**Comment by Aeramor** [comment]
Yeah, it was when sidekiq 7.0 was released. They changed redis server req from 4.0->6.0, from redis->redisclient, and on heroku need ssl verify off for redis.

**Comment by tombruijn** [comment]
Thanks for letting me know @Aeramor! I'm working on a fix.

**Comment by tombruijn** [comment]
@Aeramor I've published the fix in AppSignal for Ruby gem 3.1.6.
